src/code/z_collision_check.c

```diff
@@ -1,392 +1,3345 @@
+
+    for (col = check->colAc; col < check->colAc + check->colAcCount; col++) {
+        temp = *col;
+        if (temp != NULL) {
+            if (temp->acFlags & 1) {
+                if (temp->actor == NULL || temp->actor->update != NULL) {
+                    (*D_8011DF5C[temp->shape])(globalCtx, check, temp);
+                }
+            }
+        }
+    }
+}
+
+void CollisionCheck_AC(GlobalContext* globalCtx, CollisionCheckContext* check, Collider* collider) {
+    Collider** col;
+    Collider* temp;
+
+    static void (*D_8011DF6C[4][4])(GlobalContext*, CollisionCheckContext*, Collider*, Collider*) = {
+        { CollisionCheck_AC_JntSphVsJntSph, CollisionCheck_AC_JntSphVsCyl, CollisionCheck_AC_JntSphVsTris,
+          CollisionCheck_AC_JntSphVsQuad },
+        { CollisionCheck_AC_CylVsJntSph, CollisionCheck_AC_CylVsCyl, CollisionCheck_AC_CylVsTris,
+          CollisionCheck_AC_CylVsQuad },
+        { CollisionCheck_AC_TrisVsJntSph, CollisionCheck_AC_TrisVsCyl, CollisionCheck_AC_TrisVsTris,
+          CollisionCheck_AC_TrisVsQuad },
+        { CollisionCheck_AC_QuadVsJntSph, CollisionCheck_AC_QuadVsCyl, CollisionCheck_AC_QuadVsTris,
+          CollisionCheck_AC_QuadVsQuad },
+    };
+    for (col = check->colAc; col < check->colAc + check->colAcCount; col++) {
+        temp = *col;
+        if (temp == NULL) {
+            continue;
+        } else if (!(temp->acFlags & 1)) {
+            continue;
+        } else if (temp->actor != NULL && temp->actor->update == NULL) {
+            continue;
+        } else if (!((temp->acFlags & collider->atFlags) & 0x38)) {
+            continue;
+        } else if (collider == temp) {
+            continue;
+        }
+
+        else if ((collider->atFlags & 0x40) || collider->actor == NULL || temp->actor != collider->actor) {
+            D_8011DF6C[collider->shape][temp->shape](globalCtx, check, collider, temp);
+        }
+    }
+}
+
+// Test Attack Collisions
+void func_8006139C(GlobalContext* globalCtx, CollisionCheckContext* check) {
+    Collider** colAt;
+    Collider* colliderAt;
+
+    if (check->colAtCount != 0) {
+        if (check->colAcCount != 0) {
+            for (colAt = check->colAt; colAt < check->colAt + check->colAtCount; colAt++) {
+                colliderAt = *colAt;
+                if (colliderAt != NULL) {
+                    if (colliderAt->atFlags & 1) {
+                        if ((colliderAt->actor == NULL) || (colliderAt->actor->update != NULL)) {
+                            CollisionCheck_AC(globalCtx, check, colliderAt);
+                        }
+                    }
+                }
+            }
+            func_800611A0(globalCtx, check);
+        }
+    }
+}
+
+// Get mass type
+s32 func_8006146C(u8 mass) {
+    if (mass == 0xFF) {
+        return 0;
+    }
+    if (mass == 0xFE) {
+        return 1;
+    }
+    return 2;
+}
+
+#ifdef NON_MATCHING
+// SetOC collision, perform elastic collision
+// Regalloc issues, possibly logic issues too
```

Comment:
As mentioned in a previous comment, `NON_MATCHING` should only be used if the function is known to be completely equivalent.

Reply:
i'm more confident that this one is equivalent, so i left it as is

Reply:
I would remove the mention about possible logic issues then, it could confuse anyone looking through this code in the future.